### PR TITLE
DefaultLabelProvider: Add collapsible SideNavigation accessibility labels

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

Added type definitions for the collapsible SideNavigation default accesssibility labels.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-XXXX)
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers, relevant feature teams)

### CODE REVIEW

**Review by willsmanley — COMMENTED** [review]
don't we need defaults for `fallbackLabels` for each of these new labels?

File: packages/gestalt/dist/index.d.ts [hunk 1]
```diff
@@ -493,6 +493,9 @@ interface DefaultLabelProviderProps {
         };
         SideNavigation: {
           accessibilityDismissButtonLabel: string;
+          accessibilityCollapseButtonLabel: string;
+          accessibilityExpandButtonLabel: string;
+          accessibilityEllipsisLabel: string;
```
Comment (willsmanley) [inline packages/gestalt/dist/index.d.ts:498]: i think we need to update `packages/gestalt/src/contexts/DefaultLabelProvider.jsdom.test.js` as well

Comment (willsmanley) [reply packages/gestalt/dist/index.d.ts:498]: and docs here: `docs/examples/defaultlabelprovider/translations.js`

Comment (AlbertCarreras) [reply packages/gestalt/dist/index.d.ts:498]: Whole discussion here https://pinterest.slack.com/archives/G01GN7CE99S/p1711367607564319

**Pull request closed without merging**